xbmc/utils/log.cpp

```diff
@@ -18,199 +18,114 @@
 #include "log.h"
-#include "system.h"
-#include "threads/SingleLock.h"
-#include "threads/Thread.h"
-#include "utils/StringUtils.h"
 #include "CompileInfo.h"
+#include "filesystem/File.h"
+#include "utils/URIUtils.h"
 
-static const char* const levelNames[] =
-{"DEBUG", "INFO", "NOTICE", "WARNING", "ERROR", "SEVERE", "FATAL", "NONE"};
-
-// add 1 to level number to get index of name
-static const char* const logLevelNames[] =
-{ "LOG_LEVEL_NONE" /*-1*/, "LOG_LEVEL_NORMAL" /*0*/, "LOG_LEVEL_DEBUG" /*1*/, "LOG_LEVEL_DEBUG_FREEMEM" /*2*/ };
-
-// s_globals is used as static global with CLog global variables
-#define s_globals XBMC_GLOBAL_USE(CLog).m_globalInstance
-
-CLog::CLog()
-{}
-
-CLog::~CLog()
-{}
-
-void CLog::Close()
-{
-  CSingleLock waitLock(s_globals.critSec);
-  s_globals.m_platform.CloseLogFile();
-  s_globals.m_repeatLine.clear();
-}
-
-void CLog::Log(int loglevel, const char *format, ...)
-{
-  if (IsLogLevelLogged(loglevel))
-  {
-    va_list va;
-    va_start(va, format);
-    LogString(loglevel, StringUtils::FormatV(format, va));
-    va_end(va);
-  }
-}
-
-void CLog::LogFunction(int loglevel, const char* functionName, const char* format, ...)
-{
-  if (IsLogLevelLogged(loglevel))
-  {
-    std::string fNameStr;
-    if (functionName && functionName[0])
-      fNameStr.assign(functionName).append(": ");
-    va_list va;
-    va_start(va, format);
-    LogString(loglevel, fNameStr + StringUtils::FormatV(format, va));
-    va_end(va);
-  }
-}
+static const std::string LogFileExtension = ".log";
 
-void CLog::LogString(int logLevel, const std::string& logString)
-{
-  CSingleLock waitLock(s_globals.critSec);
-  std::string strData(logString);
-  StringUtils::TrimRight(strData);
-  if (!strData.empty())
-  {
-    if (s_globals.m_repeatLogLevel == logLevel && s_globals.m_repeatLine == strData)
-    {
-      s_globals.m_repeatCount++;
-      return;
-    }
-    else if (s_globals.m_repeatCount)
-    {
-      std::string strData2 = StringUtils::Format("Previous line repeats %d times.",
-                                                s_globals.m_repeatCount);
-      PrintDebugString(strData2);
-      WriteLogString(s_globals.m_repeatLogLevel, strData2);
-      s_globals.m_repeatCount = 0;
-    }
-
-    s_globals.m_repeatLine = strData;
-    s_globals.m_repeatLogLevel = logLevel;
+spdlog::sink_ptr CLog::m_fileSink = nullptr;
+CLog::Logger CLog::m_defaultLogger = nullptr;
+PlatformInterfaceForCLog CLog::m_platform;
 
-    PrintDebugString(strData);
-
-    WriteLogString(logLevel, strData);
-  }
-}
+int CLog::m_logLevel = LOG_LEVEL_NORMAL;
+int CLog::m_extraLogLevels = 0;
 
-bool CLog::Init(const std::string& path)
+void CLog::Initialize(const std::string& path)
 {
-  CSingleLock waitLock(s_globals.critSec);
+  if (m_fileSink != nullptr)
+    return;
 
-  // the log folder location is initialized in the CAdvancedSettings
-  // constructor and changed in CApplication::Create()
+  // the given path must not be empty
+  assert(!path.empty());
```

Comment:
the compiler will remove this for release builds. Can we instead just use stderr in this case?